core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java

```diff
@@ -185,6 +238,96 @@ public void toPlan(PlanSink sink) {
         }
     }
 
+    private static class ConstContainsStrFunction extends BooleanFunction implements UnaryFunction {
+        private final boolean caseInsensitive;
+        private final String pattern;
+        private final Function value;
+
+        public ConstContainsStrFunction(Function value, String pattern, boolean caseInsensitive) {
+            this.value = value;
+            this.pattern = pattern;
+            this.caseInsensitive = caseInsensitive;
+        }
+
+        @Override
+        public Function getArg() {
+            return value;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            CharSequence cs = getArg().getStr(rec);
+            if (cs == null) {
+                return false;
+            }
+            if (caseInsensitive) {
+                return Chars.containsIgnoreCase(cs, pattern);
```

Comment:
nit: there is enother optimisation opportunity; 
perhaps we can make assumption about the pattern and lowecase it just once rather than repitedly for each column value?

Reply:
This would limit the versatility of the newly added `Chars` functions, so I'd better keep them as is for now. Later, when we decide to cut a few hundred CPU cycles from these LIKE/ILIKE patterns, we can introduce task-specific functions in the `AbstractLikeStrFunctionFactory` class. WDYT?

Reply:
Addressed in a82a28caec8efe433062c11d2db0644bb49a954d